app/controllers/referees/build_references_controller.rb

```diff
@@ -47,7 +47,10 @@ def update
       if @form.valid?
         @reference.update!(@form.params_to_save)
         # invalidate token after reference is complete
-        if @reference.complete?
+        if !@reference.can_give_reference?
+          @reference.mark_as_declined
```

Comment:
IMHO this state transition is unnecessary - can_give_reference? already returns false if the reference is declined